tensorflow/contrib/py2tf/convert/builtin_functions.py

```diff
@@ -31,7 +32,7 @@ class BuiltinFunctionTransformer(gast.NodeTransformer):
   def _convert_len(self, node):
 
     def template(args):
-      tf.shape(args)[0]  # pylint:disable=undefined-variable,expression-not-assigned
+      tf.shape(args)[0]  # pylint:disable=expression-not-assigned
```

Comment:
The idea is all code in TF, unless they are planned as standalone binaries, should reach into tf, import only what they need and use that instead. So this code should not be using tf.shape.
It seems to be a general problem with py2tf, maybe there is something I am missing. @mdanatg to confirm

Reply:
@alexbw FYI

What I think is confusing about these blocks is that they are meant to be code templates (the `template` function is never called, only parsed into an AST). They are used to generate user-visible, hance the use of `tf`.

I can think of a number of solutions, as far as linters are concerned:

1. Add a dummy `tf` variable to trick the linter
2. Add linter-suppression code (that's what those `#pylint:disable` lines were meant for); for flake8, I believe that would be `#noqa`
3. Rewrite the templates as string constants; we lose any syntax checking that way, but things become less confusing.

(1) could be applied in this very CL, I think
(2) could also be fairly quick to implement (maybe it's a good short-term fix)
(3) would take a bit longer

Specifically, for (3) we would go from:

    def template(args):
      ...code...

to

    template = parse_template(
          args,
          """
              ...code...
          """)

Reply:
Update: we have a change incoming that replaces fixes the offending statements in py2tf using method (3) mentioned above. It should get through over the coming days.